### ISSUE TITLE
GUI: Empty cluster created on Vidua-Finches data

### ISSUE THREAD

**Issue by ssantichaivekin** [ssantichaivekin/empress#181]
Here's the link to the data used in this issue: [Vidua-Finches](https://drive.google.com/drive/folders/1X6c_PBuFRWewgtZEukoL6fAqu48eiJYO). A copy of this data is provided at the end of this issue.

**Problem:**

Empty cluster showing up on `View reconciliation space` window.
<img width="1184" alt="Screen Shot 2020-07-23 at 9 27 32 PM" src="https://user-images.githubusercontent.com/19219105/88360841-eeafae00-cd2b-11ea-9ba0-35ee29d7a39f.png">

**Steps to reproduce:**

1. Load the host/parasite/mapping of the Vidua-Finches data to empress GUI.
2. Change the costs to duplication = 1.37, transfer = 5.43, loss = 1.00
3. Click `Compute Reconciliations`. The lower three buttons become available.
4. Click `View solution space` -> `Clusters`. `Set the number of clusters` window pops up.
5. Set the number of clusters to 2 and then click on enter.
5. The second cluster is empty as shown in the image above.

This bug is discovered by Prof Ran.

Host:
```
((O_atricolis, A_subflava) h1, (((G_ianthinogaster, G_granatia) h64, ((E_dybowskii, (H_niveoguttatus, H_margaritatus) h71) h68, ((C_monteiri, ((L_larvata, (L_rara, ((L_r_cingica, L_r_rubricata) h100, (L_virata, (L_rhodopareia, L_sanguinodorsalis) h105) h101) h99) h97) h88, ((L_rufopicta, L_nitidula) h90, (L_senegala_rendalii, L_senegala_rhodopsis) h91) h89) h87) h74, ((P_melba_grotei, P_melba_citerior) h76, (P_afra, (P_lineata, (P_phoenicoptera, P_hypogrammica) h83) h81) h77) h75) h69) h65) h48, ((C_quartinia, C_melanotis) h50, ((E_astrild, (E_paludicola, (E_rhodophyga, (E_troglodytes, E_melpoda) h61) h59) h57) h54, E_erythronotos) h51) h49) h2) h0;
```
Parasite:
```
(((V_fischeri, V_regia) p6, ((((V_nigeriae, V_raricola) p118, (((V_camerunensis, V_maryae) p120, V_larvaticola) p116, V_wilsoni) p15) p12, (V_funera, (V_purpurascens, (V_chalybeata_S, V_codringtoni) p107) p21) p13) p10, V_chalybeata_W) p7) p4, ((((V_paradisaea, V_obtusa) p32, (V_orientalis, (V_interjecta, V_togoensis) p37) p33) p28, (V_macroura_S, V_macroura_W) p29) p26, V_hypocherina) p5) p3;
```
Mapping:
```
V_macroura_S:E_astrild
V_maryae:L_sanguinodorsalis
V_hypocherina:E_erythronotos
V_larvaticola:C_monteiri
V_obtusa:P_afra
V_togoensis:P_hypogrammica
V_chalybeata_S:L_senegala_rendalii
V_codringtoni:H_niveoguttatus
V_regia:G_granatia
V_fischeri:G_ianthinogaster
V_wilsoni:L_rufopicta
V_raricola:A_subflava
V_camerunensis:L_rara
V_purpurascens:L_rhodopareia
V_paradisaea:P_melba_grotei
V_macroura_W:E_melpoda
V_nigeriae:O_atricolis
V_chalybeata_W:L_senegala_rhodopsis
V_orientalis:P_melba_citerior
V_interjecta:P_phoenicoptera
V_funera:L_r_rubricata
```

**Comment by Qing-Yang-1211** [comment]
Sorry but I don't quite understand this bug. My understanding is that if we request for 2 clusters and there is only one histogram solution, there will have to be duplication and an empty cluster. However, if there are two histogram solutions as shown below then there will be no empty clusters:
<img width="1112" alt="Screen Shot 2020-07-24 at 8 45 00 PM" src="https://user-images.githubusercontent.com/54883284/88392611-de143c00-cdee-11ea-9de4-691fed046762.png">

**Comment by ssantichaivekin** [comment]
Change the costs to duplication = 1.37, transfer = 5.43, loss = 1.00.

**Comment by ssantichaivekin** [comment]
This is probably a clustering bug @aremath .

**Comment by ssantichaivekin** [comment]
Also resolved, probably by #199
